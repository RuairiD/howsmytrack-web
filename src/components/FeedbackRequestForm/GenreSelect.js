import React from "react";

import { Select } from "antd";
import GENRE_OPTIONS from "../../genres";

const GenreSelect = ({ decorator }) => decorator(
    <Select>
        {Object.keys(GENRE_OPTIONS).map((genreOption) => (
            <Select.Option key={genreOption} value={genreOption}>
                {GENRE_OPTIONS[genreOption]}
            </Select.Option>
        ))}
    </Select>,
);

export default GenreSelect;
