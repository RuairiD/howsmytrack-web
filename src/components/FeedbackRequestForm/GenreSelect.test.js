import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import GenreSelect from "./GenreSelect";

describe("GenreSelect", () => {
    it("renders a Select containing all available genre options", () => {
        const wrapper = shallow(
            <GenreSelect
                decorator={(x) => x}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
