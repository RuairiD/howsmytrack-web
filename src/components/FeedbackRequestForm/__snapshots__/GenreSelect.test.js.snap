// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreSelect renders a Select containing all available genre options 1`] = `
<Select
  choiceTransitionName="zoom"
  showSearch={false}
  transitionName="slide-up"
>
  <Option
    key="NO_GENRE"
    value="NO_GENRE"
  >
    No Genre
  </Option>
  <Option
    key="ELECTRONIC"
    value="ELECTRONIC"
  >
    Electronic
  </Option>
  <Option
    key="HIPHOP"
    value="HIPHOP"
  >
    Hip-Hop/Rap
  </Option>
  <Option
    key="JAZZ"
    value="JAZZ"
  >
    Jazz
  </Option>
  <Option
    key="POP"
    value="POP"
  >
    Pop
  </Option>
  <Option
    key="ROCK_METAL_PUNK"
    value="ROCK_METAL_PUNK"
  >
    Rock/Metal/Punk
  </Option>
  <Option
    key="URBAN_RNB"
    value="URBAN_RNB"
  >
    Urban/R&B
  </Option>
</Select>
`;
